### ISSUE TITLE
backwards for autograd.Function

### ISSUE THREAD

**Issue by t-vi** [Lightning-AI/lightning-thunder#1017]
We currently "fudge" autograd.Function by running through the forward as if it was the function.
(This is not good when there is `.detach()` or some such in the forward.)

Given that
- we do know the call args (in the sense of "meta") of the backward once we have traced through the forward,
